### PR TITLE
Update docstrings and minor refactor.

### PR DESCRIPTION
**Pull request by Jeff-Meadows**
Bump version to 3.0.0.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @Jeff-Meadows has signed the CLA. Thanks for the pull request!

File: flaky/_flaky_plugin.py [hunk 1]
```diff
@@ -23,9 +23,6 @@ def stream(self):
         Returns the stream used for building the flaky report.
         Anything written to this stream before the end of the test run
         will be written to the flaky report.
-
-        :return:
-        :rtype:
```
Comment (jmoldow) [inline flaky/_flaky_plugin.py]: Shouldn't this still have a documented return type?

File: flaky/_flaky_plugin.py [hunk 2]
```diff
@@ -100,25 +97,67 @@ def _log_intermediate_failure(self, err, flaky, name):
...
-        has_failed = self._has_flaky_test_failed(flaky)
+            return False
+        flaky_attributes = self._get_flaky_attributes(test)
+        flaky_attributes[FlakyNames.CURRENT_RUNS] += 1
+        has_failed = self._has_flaky_test_failed(flaky_attributes)
         if not has_failed:
-            return True, self._should_rerun_test(test, name, err)
-        return False, False
+            return True
+        return False
```
Comment (jmoldow) [inline flaky/_flaky_plugin.py]: The last three lines can be replaced with ``return not has_failed``.

File: flaky/_flaky_plugin.py [hunk 3]
```diff
@@ -100,25 +97,67 @@ def _log_intermediate_failure(self, err, flaky, name):
...
+        :type test:
+            :class:`nose.case.Test` or :class:`Function`
+        :param name:
+            The name of the test that has raised an error
+        :type name:
+            `unicode`
+        :param err:
+            Information about the test failure (from sys.exc_info())
+        :type err:
+            `tuple` of `class`, :class:`Exception`, `traceback`
```
Comment (jmoldow) [inline flaky/_flaky_plugin.py]: Should be ``type``, not ```class```.

File: flaky/_flaky_plugin.py [hunk 4]
```diff
@@ -138,28 +177,21 @@ def _handle_test_error_or_failure(self, test, err):
...
         if self._has_flaky_attributes(test):
             self._add_flaky_test_failure(test, err)
-            flaky = self._get_flaky_attributes(test)
-            runs = flaky[FlakyNames.CURRENT_RUNS] + 1
-            self._set_flaky_attribute(test, FlakyNames.CURRENT_RUNS, runs)
-            if need_reruns:
+            should_handle = self._should_handle_test_error_or_failure(test)
+            self._increment_flaky_attribute(test, FlakyNames.CURRENT_RUNS)
+            if should_handle:
                 flaky = self._get_flaky_attributes(test)
```
Comment (jmoldow) [inline flaky/_flaky_plugin.py]: In this function you call it `flaky`, but in a function above you renamed the same thing from `flaky` to `flaky_attributes`.

File: flaky/flaky_pytest_plugin.py [hunk 5]
```diff
@@ -54,6 +62,10 @@ def pytest_runtest_protocol(self, item, nextitem):
             py.test wrapper for the next test function to be run
         :type nextitem:
             :class:`Function`
+        :return:
+            True if no further hook implementations should be invoked.
+        :rtype:
+            `bool` or None
```
Comment (jmoldow) [inline flaky/flaky_pytest_plugin.py]: What do `False` and `None` mean? Can it be made to just be `bool`?

File: test/pytest_generate_example/conftest.py [hunk 6]
```diff
@@ -4,5 +4,8 @@
 
 
 def pytest_generate_tests(metafunc):
+    """
+    Parameterize a fixture named 'dummy_list' with a list containing one item - 'foo'
```
Comment (jmoldow) [inline test/pytest_generate_example/conftest.py]: But you just deleted foo.

**Pull request merged**